src/hooks/useElementsOverflowingIndex.ts

```diff
@@ -19,8 +31,10 @@ function useElementsOverflowingIndex({ ref, children, paddingSize, resizeDebounc
     debounceTime: resizeDebounceTime
   });
 
-  const [aggregatedChildLengths, setAggregatedChildLengths] = useState([]);
-  const [indexToSplit, setIndexToSplit] = useState(null);
+  const [aggregatedChildLengths, setAggregatedChildLengths] = useState<
+    Array<{ childLength: number; totalLength: number }>
```

Comment:
You're using this type more than once here, and it's not trivial, consider extracting:
```ts
type ChildLengths = Array<{ childLength: number; totalLength: number }>;
```